src/Core/Checkout/Cart/Rule/AffiliateCodeOfOrderRule.php

```diff
@@ -18,10 +18,12 @@ class AffiliateCodeOfOrderRule extends Rule
 
     /**
      * @internal
+     *
+     * @param array<string> $affiliateCode
      */
     public function __construct(
         protected string $operator = self::OPERATOR_EQ,
-        protected ?string $affiliateCode = null
+        protected ?array $affiliateCode = null
```

Comment:
would you mind renaming the variable so it is plural?

Reply:
Mh, then we also need to do it in the stored JSON blob in the database :thinking: I can also do that, but that would involve a much more sophisticated deprecation/migration strategy :see_no_evil: